lib/kubeclient/common.rb

```diff
@@ -310,7 +315,11 @@ def watch_pod_log(pod_name, namespace, container: nil)
     end
 
     def proxy_url(kind, name, port, namespace = '')
-      entity_name_plural = ClientMixin.pluralize_entity(kind.to_s)
+      entity_name_plural = if kind.to_s.end_with?('s')
```

Comment:
This one has to support a narrower list and plural code provides backward compatibility.
I've changed the code to be very explicit